pandas/core/groupby/groupby.py

```diff
@@ -637,10 +637,12 @@ def _selected_obj(self):
 
         if self._selection is None or isinstance(self.obj, Series):
             if self._group_selection is not None:
-                return self.obj[self._group_selection]
-            return self.obj
+                result = self.obj[self._group_selection]
+            result = self.obj
         else:
-            return self.obj[self._selection]
+            result = self.obj[self._selection]
+
+        return result.dropna() if self.dropna else result
```

Comment:
Do we need to actually drop the data itself? I thought `dropna` should just include whether we include the NA group (which the `factorize` call handled)?